esp/bindings/http/platform/httptransport.cpp

```diff
@@ -1050,6 +1080,45 @@ StringBuffer& CHttpMessage::getHeader(const char* headername, StringBuffer& head
     return headerval;       
 }
 
+bool CHttpMessage::hasHeader(const char* headername)
+{
+    if (!headername || !*headername)
+        return false;
+
+    ForEachItemIn(x, m_headers)
+    {
+        const char* header = m_headers.item(x);
+        if (header == NULL)
+            continue;
+        const char* colon = strchr(header, ':');
+        if (colon == NULL)
+            continue;
+        unsigned len = colon - header;
+        if ((strlen(headername) == len) && (strnicmp(headername, header, len) == 0))
```

Comment:
can calculate strlen(headername) once at the top since it never changes.

Reply:
Change made per suggestion.